midealocal/devices/e2/__init__.py

```diff
@@ -90,13 +90,17 @@ def _normalize_old_protocol(self, value: Any) -> OldProtocol:
             if isinstance(value, str):
                 value = OldProtocol(value)
                 if value == OldProtocol.auto:
-                    result = self.subtype <= 82 or self.subtype == 85 or self.subtype == 36353
+                    result = (
+                        self.subtype <= 82
+                        or self.subtype == 85
+                        or self.subtype == 36353
+                    )
                     value = OldProtocol.true if result else OldProtocol.false
             elif isinstance(value, int) or isinstance(value, bool):
                 value = OldProtocol.true if value else OldProtocol.false
             else:
                 raise ValueError("Invalid value for old_protocol")
-            return value
+            return cast(OldProtocol, value)

```

Comment:
This can hide mistakes. I do prefer if you use a different variable to return. Maybe called return_value instead of value.

Reply:
Indeed, you are right. Updated